manifests/charts/base/templates/zzz_profile.yaml

```diff
@@ -33,6 +33,13 @@ Finally, we can set all of that under .Values so the chart behaves without aware
 {{ fail (cat "unknown compatibility version" $.Values.compatibilityVersion) }}
 {{- end }}
 {{- end }}
+{{- if  $globals.platform }}
```

Comment:
LGTM if we change this to `platform` instead of `global.platform`.

Rationale...

Today:
* requires global: legacy gateway, istiod
* requires non-global: modern gateway, cni
* allows global, prefers non-global: ztunnel

So its a mixed bag.

Since we are moving towards global being something that you only need when composing charts, requiring a value to be set under `global` goes away from that a bit

Reply:
It's already a global [here](https://github.com/istio/istio/blob/master/operator/pkg/apis/values_types.proto#L606) - I don't want to alter that in this PR.

If we want to discuss this/make it consistent in a followup we can.

The global value is better for composition here, as `global.platform` by definition is  always be set to the same value, across all charts, whether composing or not.

If you didn't make it global, you'd have to do 


```
cni:
  platform: openshift
ztunnel:
  platform: openshift
istiod:
  platform: openshift
```

etc etc

versus 


```
global:
  platform: openshift
```

in a helm composition scenario.

Reply:
Didn't we do the 'flatten globals' work though?

Reply:
Ah but not on istiod

Reply:
Maybe I am being dense here, but it seems like with this PR you have to use `--set platform` on CNI still:

```
$ helm template cni manifests/charts/istio-cni --set global.platform=openshift | rg securitycontextconstraints
$ helm template cni manifests/charts/istio-cni --set platform=openshift | rg securitycontextconstraints
  resources: ["securitycontextconstraints"]
```

Which might mean either way we need to change a chart, its just a matter of whether it is istiod or CNI?

Reply:
The flattening should still happen after we load the platform, as should the other templates - so it shouldn't matter.

- If a chart wants globals flattened, and has a template that explicitly does this, this existing global will be flattened after we load the profiles.

- If a chart does not want globals flattened, this existing global will not be flattened after we load the profiles.

If we have a chart that expects it to be flattened but doesn't flatten it, then we need to fix that.

Reply:
My actual concern is I don't want users of ztunnel or gateway, without composition, to have to use `global`. This change would be the first that requires it.

How we get there is another question...

Reply:
Yep, it looks like `istio-cni` is broken here - it's relying on what (has been) a global value that it (doesn't) flatten.

Reply:
For now I'm making that one ref use the correct `global.platform` path - `istio-cni` already uses globals, and this value was only set via profile before (and still is).

This seems like it was just wrong before

Reply:
> My actual concern is I don't want users of ztunnel or gateway, without composition, to have to use global. This change would be the first that requires it.

Strictly speaking it doesn't require this, as people are free to not use platform profiles at all. They're just collections of discrete values people can override directly, as they have been doing until now. This is a strictly _additive_ change. With this  change, `platform` becomes an alias for some set of values people were already setting manually, and not much else. Status-quo is "continue to not use the alias and set all the values directly" - what people have been doing.

I understand this is a concern for you - can you explain or demonstrate why it is a concern in this PR specifically?

I think changing the existing global here to be a non-global would accomplish very little other than cluttering values files with duplicated values, as mentioned above - if it's not global, that just means you have to repeatedly specify it, and if it is global you do not. I see no meaningful distinction for using global vs not here, outside of that. 

- Using a global here has no effect on non-composition usage, and makes composition easier.

- Not using a global here has no effect on non-composition usage and makes composition harder.

If there's no downside to it for non-composition use cases, it makes composition easier, and it's already there as a global, what's the problem with (re)using it here?

We can make it not-global in a followup (and cause `istio-cni` to flatten globals, which it does not do today, if we really think this is valuable), but that would be a breaking change for the existing charts that use it as such, and either way not related to or dependent on what I'm doing here.

Reply:
> Using a global here has no effect on non-composition usage, and makes composition easier.

... but makes non-composition less clear, since now you need to think about whether to nest something under `global` or not. That is what I want to avoid.

The current state is that you NEVER need to set `global.X` for anything under the modern charts (ztunnel/gateway). With this change, now you have to if you want to use `platform`. Sure, you don't _need_ to use `platform`, but we have broken the consistency

Reply:
>    ... but makes non-composition less clear, since now you need to think about whether to nest something under global or not. That is what I want to avoid.

I also want to avoid that. This PR does that,  because it loads the profiles _before_ any local chart-level global descoping happens.

Which is important actually - we only collapse globals, we don't rehydrate them.

Which means if we want to set the same value across charts, we have to use the least-common-denominator - the global.

Basically, the UX options are this:

1. This PR:

`helm install  ztunnel --set global.platform=foo`
`helm install  istio-cni --set global.platform=foo`
`helm install  istiod --set global.platform=foo`
`helm install  gateway --set global.platform=foo`

1. Not using `global.platform` as the CLI override:

`helm install  ztunnel --set platform=foo`
`helm install  istio-cni --set platform=foo`
`helm install  istiod --set global.platform=foo`
`helm install  gateway --set global.platform=foo`

The former is what this PR does, because I don't want the latter situation, where we have to document people installing 5 charts, and they have to remember which charts use global scoping, and which charts do not - that's bad UX, and actually the precise problem Helm globals exist to solve.

So I think we're aligned here? What specifically do you object to?

Reply:
Ok - I still personally feel we accrue less tech debt here by just using `globals` for their intended purpose, but PTAL at https://github.com/istio/istio/pull/52981/commits/44f4d56b2fce42e305b242615046930d9634e11f

I think that will do it, it's only a little bit of extra munging, it effectively allows for the following:

```
helm install base --set global.platform=foo
helm install  ztunnel --set global.platform=foo
helm install  istio-cni --set global.platform=foo
helm install  istiod --set global.platform=foo
helm install  gateway --set global.platform=foo
```

or 

```
helm install base --set platform=foo
helm install  ztunnel --set platform=foo
helm install  istio-cni --set platform=foo
helm install  istiod --set platform=foo
helm install  gateway --set platform=foo
```